src/utils/validations/component/validations/form-with-validations.hoc.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+const ValidationsContext = React.createContext();
+
+const withValidations = (WrappedComponent) => {
+  return class extends React.Component {
+    state = {
+      errorCount: 0,
+      warningCount: 0,
+      infoCount: 0
+    }
+
+    inputs = {};
+
+    validationTypes = ['error', 'warning', 'info'];
+
+    addInput = (name, validate) => {
+      this.inputs[name] = validate;
+    }
+
+    removeInput = (name) => {
+      delete this.inputs[name];
+    }
+
+    adjustCount = (type, validationResult) => {
+      if (this.validationTypes.includes(type)) {
+        const stateProp = `${type}Count`;
+        let adjustment = -1;
+
+        if (validationResult) adjustment = 1;
+        else if (this.state[stateProp] === 0) adjustment = 0;
+
+        this.setState((prev) => { return { [stateProp]: prev[stateProp] + adjustment }; });
+      }
+    }
+
+    getContext = () => {
```

Comment:
I'd avoid the use of class properties if possible, for example this method does not need to be one - as do a couple of others.